src/main/kotlin/com/cjbooms/fabrikt/generators/controller/MicronautControllerInterfaceGenerator.kt

```diff
@@ -99,14 +94,33 @@ class MicronautControllerInterfaceGenerator(
             }
             .forEach { funcSpec.addParameter(it) }
 
+
+        // Add authentication
+        var securityOption = op.getSecurityRequirements().securityOption()
+        if(securityOption == SecurityOption.NO_SECURITY) {
+            securityOption = globalSecurity.securityOption()
+        }
+
+        if (securityOption == SecurityOption.AUTHENTICATION_REQUIRED) {
+            funcSpec.addParameter(
+            ParameterSpec.builder("authentication", MicronautImports.AUTHENTICATION)
+                .build())
+        } else if(securityOption == SecurityOption.AUTHENTICATION_OPTIONAL) {
+             funcSpec.addParameter(
+                ParameterSpec.builder("authentication", MicronautImports.AUTHENTICATION.copy(true))
+                    .build())
+        }
```

Comment:
**suggestion:** This is an example of a piece of code which I think can be simplified by adding the properties to your `enum class` as suggested in my other comment. It would then become:

```kotlin
if (securityOption.allowsAuthentication) {
  val typeName = 
    MicronautImports.AUTHENTICATION
      .copy(nullable = securityOption == SecurityOption.AUTHENTICATION_OPTIONAL)
  funcSpec.addParameter(
    ParameterSpec
      .builder("authentication", typeName)
      .build()
  )
}
```

Reply:
done with securityOption. allowsAuthorized